### ISSUE TITLE
run all tests on start/end?

### ISSUE THREAD

**Issue by mwotton** [karun012/arion#9]
It would be nice to know I'm starting from a sane place - i can run "cabal test && arion . src test" for the moment, but given that we know all the mappings already, it doesn't seem like it would be difficult. Do you have a workflow for this?

Thanks for this, btw. Really neat bit of kit, I've wanted exactly this for ages, just been too lazy to build the damn thing. Raised issues are a sign of interest & respect, not criticism :)

**Comment by karun012** [comment]
Of course. I am happy to know that there is interest. Even if you say this tool sucks I will take it as feedback and try to improve it. 

I have plans of doing that in the near future. I've tied this to [v0.1.0.7](https://github.com/karun012/arion/milestones/v0.1.0.7)

**Comment by mwotton** [comment]
there's more to think about here, too. one possible model is the idea of scopes: you are working on a particular test, so that one re-runs until it passes. then the other tests in the file run - if they pass, it can drop back to running everything. This lets you drill in and out more or less naturally.

This is hard at the moment, as we need the main entry point in each file. It may be easier with ide-backend, but it is also going to tie us to hspec more closely. Still, the model of running a file until it passes and then running everything shouldn't be tremendously difficult even in the current model. No time to pick this up myself though :/

**Comment by karun012** [comment]
We did tie to hspec :) At this point I don't know how easy it is to make it work with all test frameworks. Of course we can create one mode per framework and keep adding on to that, but do you think that is scalable? 

The other option I can think of is to make Arion a group of projects. A base that does all the monitoring and stuff. Everything else will be runners for different frameworks. More stuff to maintain in that case.

I think we should decide what route we want to take before switching this over to ide-backend.

**Comment by karun012** [comment]
:+1: for either route...

**Comment by karun012** [comment]
We should add zen-mode for this :). Zen-mode will help you remain focused on one test
